federation/pkg/federation-controller/util/meta.go

```diff
@@ -92,3 +97,16 @@ func DeepCopyApiTypeOrPanic(item interface{}) interface{} {
 	}
 	return result
 }
+
+func IsFederationOnlyObject(object runtime.Object) bool {
+	accessor, err := meta.Accessor(object)
+	if err != nil {
```

Comment:
Even if an error being returned is a good indication that the object is not federation-only, consider logging the error to aid in troubleshooting.